packages/react-query/package.json

```diff
@@ -11,15 +11,19 @@
     "url": "https://github.com/sponsors/tannerlinsley"
   },
   "type": "module",
-  "types": "build/lib/index.d.ts",
-  "main": "build/lib/index.legacy.cjs",
-  "module": "build/lib/index.legacy.js",
+  "types": "build/legacy/index.d.ts",
+  "main": "build/legacy/index.cjs",
+  "module": "build/legacy/index.js",
```

Comment:
I've run `build:all` locally on this PR, it still outputs things into `build/lib`:

```
CLI Using tsconfig: tsconfig.json
CLI tsup v7.0.0
CLI Using tsup config: /query/packages/react-query/tsup.config.js
CLI Target: chrome84,firefox90,edge84,safari15,ios15,opera70
CLI Cleaning output folder
CJS Build start
ESM Build start
CJS build/lib/suspense.cjs                    1.84 KB
CJS build/lib/types.cjs                       775.00 B
CJS build/lib/useInfiniteQuery.cjs            1.48 KB
CJS build/lib/useIsFetching.cjs               2.28 KB
CJS build/lib/useQueries.cjs                  5.08 KB
CJS build/lib/useMutation.cjs                 2.83 KB
CJS build/lib/HydrationBoundary.cjs           2.27 KB
CJS build/lib/QueryErrorResetBoundary.cjs     2.52 KB
CJS build/lib/useQuery.cjs                    1.31 KB
CJS build/lib/isRestoring.cjs                 1.96 KB
CJS build/lib/utils.cjs                       1.21 KB
CJS build/lib/QueryClientProvider.cjs         2.48 KB
CJS build/lib/useMutationState.cjs            3.23 KB
CJS build/lib/useBaseQuery.cjs                3.89 KB
CJS build/lib/queryOptions.cjs                1.13 KB
CJS build/lib/errorBoundaryUtils.cjs          2.57 KB
CJS build/lib/index.cjs                       3.29 KB
CJS build/lib/suspense.cjs.map                2.00 KB
CJS build/lib/types.cjs.map                   3.33 KB
CJS build/lib/useInfiniteQuery.cjs.map        1.21 KB
CJS build/lib/useQueries.cjs.map              11.19 KB
CJS build/lib/useIsFetching.cjs.map           1.13 KB
CJS build/lib/QueryErrorResetBoundary.cjs.map 1.77 KB
CJS build/lib/utils.cjs.map                   596.00 B
CJS build/lib/isRestoring.cjs.map             510.00 B
CJS build/lib/QueryClientProvider.cjs.map     1.53 KB
CJS build/lib/useQuery.cjs.map                1.52 KB
CJS build/lib/useMutationState.cjs.map        3.22 KB
CJS build/lib/useMutation.cjs.map             2.63 KB
CJS build/lib/HydrationBoundary.cjs.map       1.51 KB
CJS build/lib/useBaseQuery.cjs.map            4.27 KB
CJS build/lib/queryOptions.cjs.map            1.49 KB
CJS build/lib/index.cjs.map                   1.45 KB
CJS build/lib/errorBoundaryUtils.cjs.map      2.23 KB
CJS ⚡️ Build success in 83ms
ESM build/lib/errorBoundaryUtils.js          824.00 B
ESM build/lib/index.js                       1.07 KB
ESM build/lib/queryOptions.js                120.00 B
ESM build/lib/suspense.js                    743.00 B
ESM build/lib/types.js                       33.00 B
ESM build/lib/useInfiniteQuery.js            415.00 B
ESM build/lib/useIsFetching.js               619.00 B
ESM build/lib/useMutationState.js            1.47 KB
ESM build/lib/isRestoring.js                 316.00 B
ESM build/lib/utils.js                       217.00 B
ESM build/lib/HydrationBoundary.js           579.00 B
ESM build/lib/QueryClientProvider.js         772.00 B
ESM build/lib/QueryErrorResetBoundary.js     819.00 B
ESM build/lib/useQueries.js                  3.24 KB
ESM build/lib/useQuery.js                    278.00 B
ESM build/lib/useMutation.js                 1.16 KB
ESM build/lib/useBaseQuery.js                2.10 KB
ESM build/lib/errorBoundaryUtils.js.map      2.17 KB
ESM build/lib/index.js.map                   1.43 KB
ESM build/lib/queryOptions.js.map            1.45 KB
ESM build/lib/suspense.js.map                1.94 KB
ESM build/lib/types.js.map                   71.00 B
ESM build/lib/useInfiniteQuery.js.map        1.18 KB
ESM build/lib/useIsFetching.js.map           1.09 KB
ESM build/lib/utils.js.map                   553.00 B
ESM build/lib/isRestoring.js.map             456.00 B
ESM build/lib/QueryClientProvider.js.map     1.47 KB
ESM build/lib/QueryErrorResetBoundary.js.map 1.72 KB
ESM build/lib/HydrationBoundary.js.map       1.47 KB
ESM build/lib/useQueries.js.map              11.23 KB
ESM build/lib/useMutationState.js.map        3.18 KB
ESM build/lib/useQuery.js.map                1.49 KB
ESM build/lib/useMutation.js.map             2.60 KB
ESM build/lib/useBaseQuery.js.map            4.29 KB
ESM ⚡️ Build success in 92ms
```

Reply:
there are also some warnings when running the build:

```
> @tanstack/react-query@5.0.0-alpha.68 build:tsup /query/packages/react-query
> tsup

▲ [WARNING] "tsconfig.json" does not affect esbuild's own target setting [tsconfig.json]

    ../../tsconfig.json:28:4:
      28 │     "target": "ES2020"
         ╵     ~~~~~~~~
```

Reply:
ah nevermind - I hadn't pulled the latest changes 🤦 🙈